backend/shork/src/test/kotlin/compiler/TestParser.kt

```diff
@@ -0,0 +1,413 @@
+                        ),
+                    ),
+
+                    // SLT, LDP, STP get:
+                    // - AB if A-Mode is IMMEDIATE
+                    // Always B otherwise
+                    // AB if A-Mode is IMMEDIATE is handled via
+                    // generateAModeImmediateBModeAnyModifierAB
+                )
+
+            return arguments +
+                generateAlwaysModifierXXX() +
+                generateAModeImmediateBModeAnyModifierAB()
+        }
+
+        private fun generateAModeImmediateBModeAnyModifierAB(): List<Arguments> {
+            val firstAddress = 42
+            val secondAddress = 1337
+            val instructions =
+                listOf(
+                    Pair(TokenType.MOV, Mov::class),
+                    Pair(TokenType.SEQ, Seq::class),
+                    Pair(TokenType.SNE, Sne::class),
+                    Pair(TokenType.ADD, Add::class),
+                    Pair(TokenType.SUB, Sub::class),
+                    Pair(TokenType.MUL, Mul::class),
+                    Pair(TokenType.DIV, Div::class),
+                    Pair(TokenType.MOD, Mod::class),
+                    Pair(TokenType.SLT, Slt::class),
+                    Pair(TokenType.LDP, Ldp::class),
+                    Pair(TokenType.STP, Stp::class),
+                )
+            val arguments = mutableListOf<Arguments>()
+
+            for (instruction in instructions) {
+                val firstMode = Triple(AddressMode.IMMEDIATE, "#", TokenType.HASHTAG)
+                for (secondMode in addressModes) {
+                    val instance =
+                        instruction.second.constructors
+                            .first()
+                            .call(
+                                firstAddress,
+                                secondAddress,
+                                firstMode.first,
+                                secondMode.first,
+                                Modifier.AB,
+                            )
+                    arguments.add(
+                        Arguments.of(
+                            listOf(
+                                Token(instruction.first, instruction.first.toString(), "", 1),
+                                Token(firstMode.third, firstMode.second, "", 1),
+                                Token(TokenType.NUMBER, firstAddress.toString(), firstAddress, 1),
+                                Token(TokenType.COMMA, ",", "", 1),
+                                Token(secondMode.third, secondMode.second, "", 1),
+                                Token(TokenType.NUMBER, secondAddress.toString(), secondAddress, 1),
+                                Token(TokenType.EOF, "", "", 1),
+                            ),
+                            listOf(instance),
+                        )
+                    )
+                }
+            }
+
+            return arguments
+        }
+
+        private fun generateAlwaysModifierXXX(): List<Arguments> {
```

Comment:
This needs more explanation, as the function name isn't clear enough

Reply:
Already addressed :)